src/org/abymaly/code/Sieve.java

```diff
@@ -0,0 +1,45 @@
+package org.abymaly.code;
+
+public class Sieve {
+
+    public boolean[] createSieve(int maxVal) {
+
+        boolean[] sieve = new boolean[maxVal + 1];
+
+        for (int i = 0; i <= maxVal; ++i) {
+            sieve[i] = true;
+        }
+
+        sieve[0] = false;
+        sieve[1] = false;
+
+        for (int idx = 2; idx <= maxVal; ++idx) {
+            if (sieve[idx]) {
+                for (int i = idx * 2; i <= maxVal; i += idx) {
+                    sieve[i] = false;
+                }
+            }
+        }
+
+        return sieve;
+    }
+
+    public String sieveToString(boolean[] sieve) {
+
+        StringBuilder sieveToString = new StringBuilder();
+
+        for (int i = 0; i < sieve.length; ++i) {
+            sieveToString
+                    .append(sieve[i])
+                    .append("; ");
+        }
+
+        return sieveToString.toString();
+    }
+
+    public boolean isPrimary(int val) {
+        Sieve sieve = new Sieve();
```

Comment:
Niech ten newSieve dzieje się w konstruktorze klasy, niech tworzy jakąś prywatną boolean[] o arbitralnie wybranym rozmiarze,
a potem niech isPrimary nie tworzy nowego sita, tylko niech korzysta z tego już utworzonego.
A jak argument jest za duży (większy niż ta tablica o arbitralnie dużym rozmiarze), to można wyjątkiem rzucić (od tego są), albo zwrócić cokolwiek